pkg/tsdb/cloudwatch/cloudwatch_integration_test.go

```diff
@@ -239,4 +250,43 @@ func Test_CloudWatch_CallResource_Integration_Test(t *testing.T) {
 		require.Nil(t, err)
 		assert.JSONEq(t, `[{"value":{"name":"field1","percent":50}},{"value":{"name":"field2","percent":50}}]`, string(sent.Body))
 	})
+
+	t.Run("Should handle region requests and return regions from the api", func(t *testing.T) {
```

Comment:
What are we showing in this test?
(Are we showing that with the new refactored endpoint, now the returned JSON is in a different format?) What's the argument for the new format again?

Why do we only assert one of the regions is present? Why not assert the entire body?

Reply:
> What are we showing in this test

To me our integration tests are showing that when you put together a new executor it will handle requests for /regions. We are also specifying that when the feature flag is set to true, it will send back regions in a particular format. I think all of our integration tests here are mostly demonstrating like a happy path of what endpoints our backend handles, to just quickly validate that our models, services, and routes handlers all kind of work together.

> (Are we showing that with the new refactored endpoint, now the returned JSON is in a different format?) What's the argument for the new format again?
Yes.

The new format is the same as the other resource formats. Before it sent things over like this: 
![Screenshot 2023-09-19 at 12 52 24 PM](https://github.com/grafana/grafana/assets/6620164/c0562678-7aa3-42bd-8bd6-c46c8df19b89) 
to something like this:
![Screenshot 2023-09-19 at 12 55 01 PM](https://github.com/grafana/grafana/assets/6620164/27ea87c5-5d44-48d3-99eb-ac4522d5aedc)

This also follows the changes we did with log groups:
![Screenshot 2023-09-19 at 12 57 19 PM](https://github.com/grafana/grafana/assets/6620164/6fe7c15c-13df-4cd7-a38a-742b6fd4c85b)

The reasoning here is that the former is very tied to the UI element and kind of brittle as a result. The latter is hopefully a bit more flexible as ui's change, and more accurately reflects the actual data. 

I think arguably this change does not HAVE to happen now just to fix the unexpected token bug, but I think we might as well kind of change this all at once? I think it was going to be just as much work imo to do it the "right" way as it was to keep the old stuff so I felt like this seemed ok to me. 

> Why do we only assert one of the regions is present? Why not assert the entire body?
It just seemed easier to read and to write. The Regions endpoint will ultimately return different regions depending on your particular aws setup, so it seemed reasonable to me to just validate that it is returning in the format we expect.